airflow/api_fastapi/common/parameters.py

```diff
@@ -112,6 +112,34 @@ def depends(self, only_active: bool = True) -> _OnlyActiveFilter:
         return self.set_value(only_active)
 
 
+class _NoneFilter(BaseParam[bool]):
+    """check if a column is none or not."""
+
+    def __init__(self, attribute: ColumnElement, value: bool | None = None, skip_none: bool = True) -> None:
+        super().__init__(value, skip_none)
+        self.attribute: ColumnElement = attribute
+        self.value: bool | None = value
+
+    def to_orm(self, select: Select) -> Select:
+        if not self.value and self.skip_none:
+            return select
+        return select.where(self.attribute.is_(None))
+
+    def depends(self, *args: Any, **kwargs: Any) -> Self:
+        raise NotImplementedError("Use search_param_factory instead , depends is not implemented.")
+
+
+def none_param_factory(
+    attribute: ColumnElement,
+    none_filter_name: str,
+    skip_none: bool = True,
+) -> Callable[[bool | None], _NoneFilter]:
```

Comment:
I think all that could be implemented with a `is_none` operator in the `FilterOptionEnum` ? And then using the generic `filter_param_factory` ? Might limit code addition.

Reply:
Something like (in your case):
```python
        if self.filter_option == FilterOptionEnum.IS_NONE:
            if self.value == False or self.value is None:
                return select
            if self.value == True:
                return select.where(self.attribute.is_(None))
```

Also this particular enum option is only available with `bool | None` / `bool` type.

Reply:
Modified the code